tinygrad/tensor.py

```diff
@@ -499,27 +499,30 @@ def repeat(self, repeats:Sequence[int]) -> Tensor:
     final_shape = [r*s for r,s in zip(repeats, base_shape)]
     return self.reshape(new_shape).expand(expand_shape).reshape(final_shape)
 
+  def _resolve_dim(self, dim:int, *, outer:bool=False) -> int:
```

Comment:
I think this logic can be used in reduce also

Reply:
Also in flatten, unflatten and probably multiple others. I was planning to go over them one by one in the near future whenever I have time to test.

Should I instead change them all in this PR?